src/test/java/org/folio/inventory/eventhandlers/MatchAuthorityEventHandlerUnitTest.java

```diff
@@ -0,0 +1,339 @@
+        .withContentType(MATCH_PROFILE)
+        .withContent(JsonObject.mapFrom(new MatchProfile()
+          .withExistingRecordType(AUTHORITY))));
+    assertTrue(eventHandler.isEligible(eventPayload));
+  }
+
+  @Test
+  public void shouldMatchWithSubMatchByAuthorityOnHandleEventPayload(TestContext testContext) throws UnsupportedEncodingException {
+    Async async = testContext.async();
+
+    doAnswer(ans -> {
+      Consumer<Success<MultipleRecords<Authority>>> callback = ans.getArgument(2);
+      Success<MultipleRecords<Authority>> result =
+        new Success<>(new MultipleRecords<>(singletonList(createAuthority()), 1));
+      callback.accept(result);
+      return null;
+    }).when(collection)
+      .findByCql(eq(format("personalName == \"%s\" AND id == \"%s\"", PERSONAL_NAME, AUTHORITY_ID)),
+        any(PagingParameters.class), any(Consumer.class), any(Consumer.class));
+
+    EventHandler eventHandler = new MatchAuthorityEventHandler(mappingMetadataCache);
+    HashMap<String, String> context = new HashMap<>();
+    context.put(AUTHORITY.value(), JsonObject.mapFrom(createAuthority()).encode());
+    context.put("MAPPING_PARAMS", new JsonObject().encode());
+    DataImportEventPayload eventPayload = createEventPayload().withContext(context);
+
+    eventHandler.handle(eventPayload).whenComplete((updatedEventPayload, throwable) -> {
+      testContext.assertNull(throwable);
+      testContext.assertEquals(1, updatedEventPayload.getEventsChain().size());
+      testContext.assertEquals(
+        updatedEventPayload.getEventsChain(),
+        singletonList(DI_SRS_MARC_AUTHORITY_RECORD_CREATED.value())
+      );
+      testContext.assertEquals(DI_INVENTORY_AUTHORITY_MATCHED.value(), updatedEventPayload.getEventType());
+      async.complete();
+    });
+
+  }
+
+  private Authority createAuthority() {
+    return new Authority().withId(AUTHORITY_ID).withPersonalName(PERSONAL_NAME);
+  }
+
+  private DataImportEventPayload createEventPayload() {
+    return new DataImportEventPayload()
+      .withEventType(DI_SRS_MARC_AUTHORITY_RECORD_CREATED.value())
+      .withJobExecutionId(UUID.randomUUID().toString())
+      .withEventsChain(new ArrayList<>())
+      .withOkapiUrl("http://localhost:00001")
+      .withTenant("diku")
+      .withToken("token")
+      .withContext(new HashMap<>())
+      .withCurrentNode(new ProfileSnapshotWrapper()
+        .withId(UUID.randomUUID().toString())
+        .withContentType(MATCH_PROFILE)
+        .withContent(new MatchProfile()
+          .withExistingRecordType(AUTHORITY)
+          .withIncomingRecordType(MARC_AUTHORITY)
+          .withMatchDetails(singletonList(new MatchDetail()
+            .withMatchCriterion(EXACTLY_MATCHES)
+            .withExistingMatchExpression(new MatchExpression()
+              .withDataValueType(VALUE_FROM_RECORD)
+              .withFields(singletonList(
+                new Field().withLabel("field").withValue("authority.personalName"))
```

Comment:
Please create tests for matching that is required by a feature: by 001 and 999ffs tags.

Reply:
Added 2 additional tests especially for 001 and 999s

Reply:
https://github.com/folio-org/mod-inventory/pull/448/files#diff-3606faa27d6c0c23a42883116c5dc99f806a373fa5c6fd7c4523417d73b3f16fR145